solutions/src/main/java/ru/sberbank/school/task13/cache/CacheProxy.java

```diff
@@ -0,0 +1,334 @@
+    }
+
+    private static String getCheckedPath(File directory, String directoryPath) {
+        String path;
+        if (directory.exists() && directory.isDirectory()) {
+            path = directoryPath;
+        } else {
+            boolean created = directory.mkdir();
+            if (created) {
+                path = directoryPath;
+            } else {
+                throw new IllegalArgumentException("error! directory can not be created");
+            }
+        }
+        return path;
+    }
+
+    public Object cache(Object service) {
+        Class c1ass = service.getClass();
+        return Proxy.newProxyInstance(c1ass.getClassLoader(),
+                c1ass.getInterfaces(),
+                new CacheProxyHandler(service));
+    }
+
+    private class CacheProxyHandler implements InvocationHandler {
+        private Object service;
+        private Map<String, Object> cacheMap;
+
+        CacheProxyHandler(Object service) {
+            this.service = service;
+            this.cacheMap = new HashMap<>();
+        }
+
+        @Override
+        public Object invoke(Object proxy, Method method, Object[] args)
+                throws InvocationTargetException, IllegalAccessException {
+
+            Object result = null;
+            Cache cacheAnnotation = method.getAnnotation(Cache.class);
+
+            if (!handlerInvokeMethodArgsCheck(cacheAnnotation, method, args)) {
+                return method.invoke(service, args);
+            }
+
+            CacheType cacheType = cacheAnnotation.cacheType();
+
+            if (cacheType.equals(CacheType.JVM)) {
+                result = invokeJVM(method, args);
+            } else if (cacheType.equals(CacheType.FILE)) {
+                result = invokeFile(method, args);
+            } else {
+                throw new CacheProxyException("unknown Cache type is declared in " + method.getName());
+            }
+
+            return result;
+        }
+
+        private boolean handlerInvokeMethodArgsCheck(Annotation cacheAnnotation, Method method, Object[] args) {
+            if (cacheAnnotation == null) {
+                return false;
+            }
+
+            if (method.getReturnType().equals(void.class)) {
+                throw new IllegalArgumentException("called method can not be void");
+            }
+
+            return true;
+        }
+
+        private Object invokeJVM(Method method, Object[] args) throws InvocationTargetException,
+                IllegalAccessException {
+
+            String keyFromInput = getKeyToCache(method, args);
+
+            Object result = findInCache(keyFromInput);
+
+            if (result == null) {
+                result = method.invoke(service, args);
+                cacheMap.put(keyFromInput, result);
+            }
+
+            return result;
+        }
+
+        private Object invokeFile(Method method, Object[] args) throws InvocationTargetException, IllegalAccessException {
```

Comment:
Line is longer than 120 characters (found 122).

*<this is a auto generated comment from violation-comments-lib F7F8ASD8123FSDF>* *<a-1926110076>*